### PR TITLE
fix multiple vulnerabilities or bugs (rebase for #11608)

### PR DESCRIPTION
**Pull request by Young-X**
fix multiple vulnerabilities or bugs (rebase for #11608)

### CODE REVIEW

**Comment by pelijah** [comment]
It is not good approach to use your master branch to propose changes. Check this: https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell

**Comment by radare** [comment]
760 reds

**Comment by radare** [comment]
conflicts out there, rebase pls

**Comment by radare** [comment]
cam you close the other pr if this is dupe?

**Comment by radare** [comment]
Rebase != merge. And its stil red

**Comment by Young-X** [comment]
I have messed things up. should i make a new pr? or somethings to fix up?

**Comment by radare** [comment]
You can just reuse this pr. Push -f and will be fine

**Comment by radare** [comment]
Please squash and rebase

File: libr/anal/var.c [hunk 1]
```diff
@@ -192,6 +192,9 @@ R_API int r_anal_var_retype(RAnal *a, ut64 addr, int scope, int delta, char kind
 }
 
 R_API int r_anal_var_delete_all(RAnal *a, ut64 addr, const char kind) {
+	if (!a){
```
Comment (radare) [inline libr/anal/var.c]: use the r_return_if api instead

File: libr/asm/p/asm_hexagon_gnu.c [hunk 2]
```diff
@@ -35,6 +35,7 @@ static void memory_error_func(int status, bfd_vma memaddr, struct disassemble_in
 DECLARE_GENERIC_PRINT_ADDRESS_FUNC()
 DECLARE_GENERIC_FPRINTF_FUNC()
 
+
```
Comment (radare) [inline libr/asm/p/asm_hexagon_gnu.c]: why

**Comment by radare** [comment]
hello?

**Comment by radare** [comment]
can you. please rebase and. update this pr?

**Comment by Young-X** [comment]
> 
> 
> can you. please rebase and. update this pr?

I am sorry. I am a little busy recently. I will update the pr in the next few days as soon as possible.

**Comment by radare** [comment]
Thanks! feel free to squash and rebase to cleanup the changes. Looking forward the update

**Comment by Young-X** [comment]
I have messed things up. I created a new branch and made pr again. soooooo sorry.

**Comment by radare** [comment]
Seems to introduce a bunch of regressions. run the tests in local because from the output i see in travis it seems like its segfaulting

File: libr/anal/var.c (hunk 1 above)
Comment (radare) [reply libr/anal/var.c]: use r2i to indent properly

**Comment by radare** [comment]
ping

**Comment by radare** [comment]
PING

**Pull request closed without merging**